wrappers/huaweicloudkms/huaweicloudkms.go

```diff
@@ -0,0 +1,337 @@
+package huaweicloudkms
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"sync/atomic"
+
+	"github.com/hashicorp/go-cleanhttp"
+	wrapping "github.com/hashicorp/go-kms-wrapping"
+	"github.com/huaweicloud/golangsdk"
+	huaweisdk "github.com/huaweicloud/golangsdk/openstack"
+	kmsKeys "github.com/huaweicloud/golangsdk/openstack/kms/v1/keys"
+)
+
+// These constants contain the accepted env vars; the Vault one is for backwards compat
+const (
+	EnvHuaweiCloudKMSWrapperKeyID   = "HUAWEICLOUDKMS_WRAPPER_KEY_ID"
+	EnvVaultHuaweiCloudKMSSealKeyID = "VAULT_HUAWEICLOUDKMS_SEAL_KEY_ID"
+)
+
+// Wrapper is a Wrapper that uses HuaweiCloud's KMS
+type Wrapper struct {
+	client       kmsClient
+	keyID        string
+	currentKeyID *atomic.Value
```

Comment:
I'm not sure about this currentKeyID mechanism. If this value can be changed on every encrypt operation, then it seems like any time you do a KeyID() call, this value might have been overwritten multiple times. In other words, I'm not sure it's meaningful to display the current actual value as opposed to the original alias.

Reply:
Other wrappers also change the `currentKeyID` every time the encrypt operation happens. See [aws](https://github.com/hashicorp/go-kms-wrapping/blob/master/wrappers/awskms/awskms.go#L187-L194). Actually, the key id will not change for huawei cloud kms encryption operation. In other words, the input key id is same as the one the encrypt api returns.

Reply:
In that case, why not just remove this mechanism entirely and return the input key id?

Reply:
Never mind, this is fine (see https://github.com/hashicorp/go-kms-wrapping/pull/8/files)

Reply:
Thanks!